lib/Controller/WopiController.php

```diff
@@ -122,28 +120,64 @@ public function __construct(
 	 * and general information about the capabilities that the WOPI host has on the file.
 	 */
 	public function wopiCheckFileInfo(string $documentId): JSONResponse {
-		$token = $this->request->getParam('access_token');
+		$wopiToken = $this->request->getParam('access_token');
 
-		list($fileId, , $version, $sessionId) = Helper::parseDocumentId($documentId);
-		$this->logger->info('CheckFileInfo: Getting info about file {fileId}, version {version} by token {token}.', [
+		$this->logger->debug('CheckFileInfo: documentId {documentId}.', [
 			'app' => $this->appName,
-			'fileId' => $fileId,
-			'version' => $version,
-			'token' => $token ]);
-
-		$row = new Db\Wopi();
-		$row->loadBy('token', $token);
+			'documentId' => $documentId
+		]);
 
-		$res = $row->getWopiForToken($token);
+		$res = $this->getWopiInfoForToken($documentId, $wopiToken);
 		if (!$res) {
 			$this->logger->debug('CheckFileInfo: get token failed.', ['app' => $this->appName]);
-			return new JSONResponse([], Http::STATUS_NOT_FOUND);
+			return new JSONResponse([], Http::STATUS_FORBIDDEN);
+		}
+
+		// get origin
+		$postMessageOrigin = $res['server_host'];
+
+		// get owner info
+		$ownerId = $res['owner'];
+
+		// get user info
+		if ($res['editor'] && $res['editor'] !== '' && !($res['attributes'] & WOPI::ATTR_FEDERATED)) {
```

Comment:
The `$res['editor'] && $res['editor'] !== ''` seems confusing. It seems `$res['editor']` is evaluated as a boolean even if it isn't one. This could backfire in special cases such as `$res['editor'] = '0'`

I also assume that the editor is mandatory, so in that case I don't know why we're checking `$res['editor']`

Reply:
yes, you are right, editor is always at least empty string. I did not notice. thanks!